src/Microsoft.Azure.WebJobs.Host/Hosting/FunctionDataCache/IFunctionDataCache.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// Interface to interact with the cache that contains binding data objects.
+    /// </summary>
+    public interface IFunctionDataCache : IDisposable
+    {
+        /// <summary>
+        /// Gets if the FunctionDataCache is enabled or not.
+        /// </summary>
+        bool IsEnabled { get; }
+
+        /// <summary>
+        /// Put a given object (backed by shared memory) into the cache.
+        /// </summary>
+        /// <param name="cacheKey">Key corresponding to the object being inserted into the cache.</param>
+        /// <param name="sharedMemoryMeta">Metadata about the shared memory region containing content of the object.</param>
+        /// <param name="isIncrementActiveReference">If <see cref="true"/>, the reference counter for this object in the cache will be incremented (hence preventing it from being evicted).</param>
```

Comment:
name `isIncrementActiveReference` is a little confusing. Should this be `incrementReferenceCount` or `incrementActiveReferenceCount`